_posts/2018-11-29-template-store.md

```diff
@@ -0,0 +1,167 @@
+
+* `faas-cli build`
+
+If you have multiple functions then you can also build them in parallel with the `--parallel` flag.
+
+### How it works
+
+If you run `faas-cli up` then your code will be built, pushed to a Docker registry and then deployed via the OpenFaaS RESTful API.
+
+![](/images/template-store/how-it-works.png)
+
+*Conceptual diagram showing the `faas` or `faas-cli` packaging a function*
+
+### Building the template store
+
+The template store was introduced to make it easy to build, share and make use of templates written for your favourite programming languages.
+
+Around 12 months ago today we announced the [OpenFaaS Function Store](https://blog.alexellis.io/announcing-function-store/) which was built to enable sharing and re-use of functions between our users and community. The template store builds on that success and uses a similar set of new CLI commands.
+
+<blockquote class="twitter-tweet" data-lang="en"><p lang="en" dir="ltr">In 42 seconds <a href="https://twitter.com/alexellisuk?ref_src=twsrc%5Etfw">@alexellisuk</a> demos the most powerful feature of FaaS. The function store. This is what the future and the now looks like. An open source ecosystem of functions. <a href="https://t.co/ix3ER4b7Jn">pic.twitter.com/ix3ER4b7Jn</a></p>&mdash; Kenny Bastani (@kennybastani) <a href="https://twitter.com/kennybastani/status/1064881269153116167?ref_src=twsrc%5Etfw">November 20, 2018</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script> 
+
+The OpenFaaS core contributors provide support for the *official* templates in the [openfaas/templates](https://github.com/openfaas/templates/) GitHub repo, but there are a growing number of third-party, experimental or opinionated templates which we wanted to make available without breaking the existing experience.
+
+Examples of the community's favourite templates include: [Node.js 10 with Express.js](https://github.com/openfaas-incubator/node10-express-template), [Python 2.7 & 3 with Flask](https://github.com/openfaas-incubator/python-flask-template) and until recently the unofficial PHP template. We've now merged the PHP template into the [official set](https://github.com/openfaas/templates/).
+
+### Get your template into the store
+
+The function store and the template store both have a similar model where a JSON served from the GitHub CDN provide an index of the items available to you. 
+
+> You can view the manifest file here: [templates.json](https://github.com/openfaas/store/blob/master/templates.json)
+
+To get your template into the store follow this process:
+
+* 1. Propose the template through a new issue
+
+We'll review the repo, Dockerfile and how the template is structured to make sure it is suitable for and workable in the store. You may get some feedback from the community before being given the go-ahead
+
+* 2. You'll receive the go-ahead
+
+Once you've got the go-ahead you can send a PR to the store.json file
```

Comment:
PR would be against the `template.json` file rather than the x86 function store file.